Lib/ClassLibraryCommon/File/CloudFile.cs

```diff
@@ -4937,8 +4971,77 @@ private RESTCommand<string> StartCopyImpl(Uri source, AccessCondition sourceAcce
             {
                 StorageRequestMessage msg = FileHttpRequestMessageFactory.CopyFrom(uri, serverTimeout, source, sourceAccessCondition, destAccessCondition, cnt, ctx, this.ServiceClient.GetCanonicalizer(), this.ServiceClient.Credentials);
                 FileHttpRequestMessageFactory.AddMetadata(msg, attributes.Metadata);
+
+                FileCopyOptions fileCopyOptions =
+                    null != copyOptions ?
+                        copyOptions :
+                        new FileCopyOptions()
```

Comment:
If we change `FileCopyOptions` to be a `struct` then that won't be necessary.

Reply:
Here it needs to be a nullable type, this way if customer doesn't want to set it, customer can directly pass null as a parameter..

Reply:
If you change `FileCopyOptions` to be a `struct` and define as parameter in overloads as `FileCopyOptions copyOptions = default` you should achieve same result as this code.